proxy/coco/src/source.rs

```diff
@@ -512,9 +512,35 @@ pub struct LocalState {
 ///
 /// Will return [`Error`] if the repository doesn't exist.
 pub fn local_state(repo_path: &str) -> Result<LocalState, Error> {
+    let repo = git2::Repository::open(repo_path)?;
+    let first_branch = repo
+        .branches(Some(git2::BranchType::Local))?
+        .filter_map(|branch_result| {
+            let branch = match branch_result {
+                Ok(s) => s,
+                Err(_) => return None,
+            };
+
+            match branch.0.name() {
+                Ok(Some(n)) => Some(n.to_owned()),
+                Ok(None) | Err(_) => None,
+            }
+        })
+        .min()
+        .expect("Could not find any branches.");
+
+    log::debug!(
+        "The fallback branch for this repository is: {:?}",
+        first_branch
+    );
+
     let repo = git::Repository::new(repo_path)?;
-    // TODO(finto): This should be the default branch of the project, possibly.
-    let browser = Browser::new(&repo, git::Branch::local("master"))?;
+
+    let browser = match Browser::new(&repo, git::Branch::local("master")) {
+        Ok(browser) => browser,
+        Err(_) => Browser::new(&repo, git::Branch::local(&first_branch))?,
```

Comment:
I can see how this works out but should we always treat it as an `Error` when someone doesn't use `master` as the default branch?

Reply:
When a user creates a project off of an existing repository, it's up to them to choose what the default branch should be. This request returns a list of all the branches of a repository (regardless whether it is already managed by radicle or not). We get all branches of a project via the surf `Browser`, but the API of `Browser` requires us to initialise it with a branch.

Now, we try to initialise it with "master" first and if that fails, we try with the first branch in the list, if there are no branches at all, the whole request should fail.

We could theoretically just use `Git2` for getting the branch list and not bother with `Browser`, but then we would need to add code here to determine whether the repo is already managed by radicle.